reducers/plugins/index.js

```diff
@@ -136,10 +168,18 @@ function components(state = {}, action) {
     case ActionTypes.RECEIVED_PLUGIN_COMPONENT: {
         if (action.name && action.data) {
             const nextState = {...state};
-            const currentArray = nextState[action.name] || [];
-            const nextArray = [...currentArray];
+            const currentArray = (nextState[action.name] || []).filter((c) => !c.subMenu);
+            let actionData = action.data;
+            if (action.name === 'PostDropdownMenu' && actionData.parentMenuId) {
+                const subMenu = (nextState[action.name] || []).find((c) => c.subMenu && c.pluginId === actionData.pluginId);
```

Comment:
@scottleedavis i don't think this is quite right, [`.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) will find the *first* component that matches the condition `c.subMenu && c.pluginId === actionData.pluginId`, which means if a plugin has more than 1 menu registered that has a submenu, it'll only be updating the first, which is not what we want. Unless maybe i missed something 🤔

Reply:
What we need to do is:

1. Get all the components that has a `subMenu` field and belongs to *this* pluginId.
2. Iterate over all those components and append `action.data` submenu to the correct submenu
3. If no submenu was updated because we couldn't find the correct `parentMenuId` then we should return the original unaltered state.

Reply:
Great ideas!  This is what the code currently looks like following your bullet points
```javascript
            if (action.name === 'PostDropdownMenu' && actionData.parentMenuId) {
                const subMenus = (nextState[action.name] || []).filter((c) => c.subMenu && c.pluginId === actionData.pluginId);
                const subMenu = subMenus.find(sm => hasSubmenu(sm, actionData));
                if (!subMenu) {
                    return state;
                }
                actionData = buildSubMenu(subMenu, actionData);
                if (!actionData) {
                    return state;
                }
            }
```